### PR TITLE
fix: support 0 as an explicit status code

### PR DESCRIPTION
**Pull request by tanem**
Bumped into this in MSW, setting `ctx.status(0)` results in a `200` status in the mocked response. Guessing this change will more closely match the intent of the code since `status` is a `number`?

### CODE REVIEW

**Comment by kettanaito** [comment]
Hey, @tanem. Thanks for opening this.

Could you tell me more about your use case? To my best knowledge, HTTP status codes must be within 100-5xx range in order to comply with the spec. While `ctx.status(0)` results in status code 200, I'd argue that the mistake here is setting the status code to 0.

**Comment by tanem** [comment]
Hey @kettanaito 👋 

I'm migrating tests for a legacy app. We see this scenario reported in our monitoring tools and want it handled in a specific way. It's for these sorts of scenarios described in [XMLHttpRequest.status](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/status):

> Before the request completes, the value of status is 0. Browsers also report a status of 0 in case of XMLHttpRequest errors.

Also described [here](https://www.ibm.com/support/pages/what-causes-status-code-0-indicating-reqcancelled).

Yep agree with you, it's a bit ham-fisted and I'm guessing there's a better way of testing this... but I noticed I can successfully set `ctx.status(1)` in MSW which is also outside the 100-5xx range, so figured if that range wasn't enforced and the type is `number` then `0` should be allowed too 😅 

Also wasn't sure if something needed to change elsewhere (e.g. should `UNSENT` have status `0` [here](https://github.com/mswjs/interceptors/blob/main/src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts#L132=)?), so thought I'd raise a PR for this lo-fi option to start with.

**Review by kettanaito — APPROVED** [review]

File: src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts [hunk 1]
```diff
@@ -304,7 +304,7 @@ export const createXMLHttpRequestOverride = (
           // Trigger a loadstart event to indicate the initialization of the fetch.
           this.trigger('loadstart')
 
-          this.status = mockedResponse.status || 200
+          this.status = mockedResponse.status ?? 200
```
Comment (kettanaito) [inline src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts:307]: We need to add an integration test for mocking a status code that'd fall under this change. Mocking `0` status code may be a good example.

**Comment by tanem** [comment]
> What do you think about this roadmap? Would you have the time/interest to make it happen together?

Sounds good, would love to help out 👍 Did you want the changes made under this PR, or a new PR/PRs?

**Comment by kettanaito** [comment]
I think your changes are great and we can ship them straight away. I've opened #281 to follow up whenever you're interested. Thanks for the help!

**Comment by kettanaito** [comment]
## Released: v0.17.4 🎉

This has been released in v0.17.4!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.17.4)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.17.4)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Comment by tanem** [comment]
> I think your changes are great and we can ship them straight away. I've opened #281 to follow up whenever you're interested. Thanks for the help!

No worries and thanks for the merge 🙏 Will hit that issue up if I get the chance to make those other changes.

**Pull request merged**